heron/api/src/java/com/twitter/heron/api/topology/IStatefulComponent.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.api.topology;
+
+import com.twitter.heron.api.state.State;
+
+/**
+ * Common methods for all possible components in a topology. This interface is used
+ * when defining topologies using the Java API.
+ */
+public interface IStatefulComponent extends IComponent {
```

Comment:
@nlu90 Heron feeds the state during `void initState(State state);` so it always has the reference for the `state`.